sdk/core/System.ClientModel/src/Convenience/ClientResultException.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.ClientModel.Primitives;
+using System.Globalization;
+using System.Runtime.Serialization;
+using System.Text;
+
+namespace System.ClientModel;
+
+[Serializable]
+public class ClientResultException : Exception, ISerializable
+{
+    private const string DefaultMessage = "Service request failed.";
+
+    private readonly PipelineResponse? _response;
+    private int _status;
+
+    /// <summary>
+    /// Gets the HTTP status code of the response. Returns. <code>0</code> if response was not received.
+    /// </summary>
+    public int Status
+    {
+        get => _status;
+        protected set => _status = value;
+    }
+
+    public ClientResultException(PipelineResponse response, Exception? innerException = default)
+        : base(GetMessage(response), innerException)
+    {
+        ClientUtilities.AssertNotNull(response, nameof(response));
+
+        _response = response;
+        _status = response.Status;
+    }
+
+    public ClientResultException(string message, PipelineResponse? response = default, Exception? innerException = default)
+        : base(message ?? DefaultMessage, innerException)
+    {
+        _response = response;
+        _status = response?.Status ?? 0;
+    }
+
+    /// <summary>
+    /// TBD
+    /// </summary>
+    /// <param name="info"></param>
+    /// <param name="context"></param>
```

Comment:
Fill these out?

Reply:
I have been tracking these with:
- https://github.com/Azure/azure-sdk-for-net/issues/40758
- https://github.com/Azure/azure-sdk-for-net/issues/41044

since the APIs are still in flux to conserve time possibly documenting irrelevant APIs.

But, if the group feels they are a criterion for merging to main, I will add them to this PR.  Let me know!

Reply:
Resolving for now per https://github.com/Azure/azure-sdk-for-net/pull/41016#discussion_r1443403609, but shout if you'd rather I didn't!